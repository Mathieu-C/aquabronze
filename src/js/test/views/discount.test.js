import cartService from "../../services/cart";
import discountView from "../../views/discount";
import cartItems from "../fixtures/cartItems";

beforeEach(() => {
	cartService.clear();
});

describe("discount view", () => {
	describe("render", () => {
		it("doesn't render when there is no active discount", () => {
			const domTarget = document.querySelector("#js-cart-discounts");
			discountView.render();
			expect(domTarget.innerHTML).toBe("");
		});

		it("reflects the active discounts", () => {
			cartService.populate(cartItems);
			const domTarget = document.querySelector("#js-cart-discounts");
			discountView.render();
			expect(domTarget.innerHTML).toMatchSnapshot();
		});

		it("handles multiple active discounts", () => {
			cartService.populate(cartItems);
			cartService.modifyQuantity("NARU", 5);
			const domTarget = document.querySelector("#js-cart-discounts");
			discountView.render();
			expect(domTarget.innerHTML).toMatchSnapshot();
		});
	});
});
