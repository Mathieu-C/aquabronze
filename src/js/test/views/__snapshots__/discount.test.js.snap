// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`discount view render handles multiple active discounts 1`] = `
"
      <li class=\\"discounts__item\\">
        <span>2x1 Goku POP offer</span>
        <span>-5 €</span>
      </li>
    
      <li class=\\"discounts__item\\">
        <span>x6 Naruto POP offer</span>
        <span>-6 €</span>
      </li>
    "
`;

exports[`discount view render reflects the active discounts 1`] = `
"
      <li class=\\"discounts__item\\">
        <span>2x1 Goku POP offer</span>
        <span>-5 €</span>
      </li>
    "
`;
